### PR TITLE
[dev] Add patches for WASM support

### PR DESCRIPTION
**Pull request by IsabelParedes**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -24,6 +24,9 @@ cmake -G Ninja \
     -DLLVM_LIT_ARGS=-v \
     -DLLVM_CMAKE_DIR=$PREFIX/lib/cmake/llvm \
     -DLLVM_DIR=$PREFIX/lib/cmake/llvm \
+    -DLLVM_DEFAULT_TARGET_TRIPLE=wasm32-unknown-emscripten \
+    -DLLVM_TARGETS_TO_BUILD=WebAssembly \
+    -DLLVM_ENABLE_PROJECTS="clang;flang;mlir" \
```
Comment (h-vetinari) [inline recipe/build.sh:29]: To understand, you want to build a compiler that runs on linux and compiles for emscripten? Not sure if our clang already supports that target; in any case, building clang/flang/mlir in the same feedstock will almost certainly time out.

We'll need to come up with a larger-scale plan for how we want to support wasm/emscripten in conda-forge, but as an experiment, I could imagine pushing this into a separate label `conda-forge/label/llvm_<...>`

Comment (IsabelParedes) [reply recipe/build.sh:29]: >  you want to build a compiler that runs on linux and compiles for emscripten?

Exactly.

I don't think it's necessary to rebuild clang. Emscripten uses clang internally.

For pushing to a new label, could I make the changes in this PR or should that be separate?

**Comment by IsabelParedes** [comment]
@conda-forge-admin, please rerender

**Comment by IsabelParedes** [comment]
Hi! I'm not sure why this is failing when it passed before and there are no changes...
Anyhow, closing this for now to create a new flang variant in a separate PR.

**Comment by h-vetinari** [comment]
The linux failures I'm not sure about, but the windows builds time out on this feedstock quite regularly unfortunately.

**Pull request closed without merging**